### ISSUE TITLE
scylla_setup failing in artifacts test on Ubuntu 22.04 LTS

### ISSUE THREAD

**Issue by KnifeyMoloko** [scylladb/scylla#10669]
## Installation details

Test: artifacts_test.ArtifactsTest.test_scylla_service
Build number: 40
Backend: gce: us-east1
Kernel version: 5.15.0-1004-gcp
Test-id: 0d493f3f-2084-4689-b59b-783aeab5ff8c
Start time: 2022-05-26 11:11:08
End time: 2022-05-26 11:15:12
Started by user: maciej.cisowski
Private repo: http://downloads.scylladb.com/unstable/scylla/master/deb/unified/2022-05-20T03:58:10Z/scylladb-master/scylla.list

OS / Image: `https://www.googleapis.com/compute/v1/projects/ubuntu-os-cloud/global/images/family/ubuntu-2204-lts` (gce: us-east1)

## System under test
ScyllaDB version: 5.1.dev-0.20220520.f873806c7c98 with build-id 4fc0851b832ca3a066f62c0e688291210729caa8
ScyllaDB repo: http://downloads.scylladb.com/unstable/scylla/master/deb/unified/2022-05-20T03:58:10Z/scylladb-master/scylla.list
Actual gce image used: ubuntu-2204-jammy-v20220506
AWS region: us-east1
Instance type: n1-standard-2
Scylla packages installed:
scylla 5.1.dev-0.20220520.f873806c7c98-1
scylla-conf 5.1.dev-0.20220520.f873806c7c98-1
scylla-jmx 5.1.dev-0.20220520.f873806c7c98-1
scylla-kernel-conf 5.1.dev-0.20220520.f873806c7c98-1
scylla-node-exporter 5.1.dev-0.20220520.f873806c7c98-1
scylla-python3 5.1.dev-0.20220520.f873806c7c98-1
scylla-server 5.1.dev-0.20220520.f873806c7c98-1
scylla-tools 5.1.dev-0.20220520.f873806c7c98-1
scylla-tools-core 5.1.dev-0.20220520.f873806c7c98-1

Test config file(s):

- [ubuntu2204.yaml](https://github.com/KnifeyMoloko/scylla-cluster-tests/blob/add-ubuntu22-artifacts-test/test-cases/artifacts/ubuntu2204.yaml)

## Issue description

**>>>>>>>**
We've introduced out artifacts tests on Ubuntu 22.04. During `scylla_setup` we're hitting the following error:

```
2022-05-26 11:14:52.514: (TestFrameworkEvent Severity.ERROR) period_type=one-time event_id=506e09de-71bb-472a-8687-e5c6373829f5, source=ArtifactsTest.SetUp()
exception=[Node artifacts-ubuntu2204-jenkins-db-node-0d493f3f-0-1 [34.148.81.88 | 10.142.0.93] (seed: True)] NodeSetupFailed: Encountered a bad command exit code!
Command: 'sudo /usr/lib/scylla/scylla_setup --nic ens5 --disks /dev/nvme0n1 --setup-nic-and-disks  --swap-directory / '
Exit code: 1
Stdout:
#4  0x000055cb6c130afd execute_command_internal (bash + 0x4dafd)
#5  0x000055cb6c133158 execute_command (bash + 0x50158)
#6  0x000055cb6c12438b reader_loop (bash + 0x4138b)
#7  0x000055cb6c115c06 main (bash + 0x32c06)
#8  0x00007f891239dd90 n/a (libc.so.6 + 0x29d90)
#9  0x00007f891239de40 __libc_start_main (libc.so.6 + 0x29e40)
#10 0x000055cb6c115ed5 _start (bash + 0x32ed5)
Does not able to detect coredump file, failed to configure systemd-coredump.
coredump setup failed.
Stderr:
INFO  2022-05-26 11:14:32,107 [shard 0] seastar - Created fair group io-queue-0, capacity rate 2147483:2147483, limit 12582912, rate 16777216 (factor 1), threshold 2000
INFO  2022-05-26 11:14:32,107 [shard 0] seastar - Created io group dev(0), length limit 4194304:4194304, rate 2147483647:2147483647
INFO  2022-05-26 11:14:32,107 [shard 0] seastar - Created io queue dev(0) capacities: 512:2000/2000 1024:3000/3000 2048:5000/5000 4096:9000/9000 8192:17000/17000 16384:33000/33000 32768:65000/65000 65536:129000/129000 131072:257000/257000
WARN  2022-05-26 11:14:32,187 [shard 0] iotune - Available space on filesystem at /var/tmp/mnt: 121 MB: is less than recommended: 10 GB
INFO  2022-05-26 11:14:32,187 [shard 0] iotune - /var/tmp/mnt passed sanity checks
Created symlink /etc/systemd/system/multi-user.target.wants/scylla-server.service → /lib/systemd/system/scylla-server.service.
Created symlink /etc/systemd/system/multi-user.target.wants/var-lib-scylla.mount → /etc/systemd/system/var-lib-scylla.mount.
Created symlink /etc/systemd/system/multi-user.target.wants/var-lib-systemd-coredump.mount → /etc/systemd/system/var-lib-systemd-coredump.mount.
Traceback (most recent call last):
File "/jenkins/slave/workspace/scylla-staging/maciej/artifacts-ubuntu/scylla-cluster-tests/sdcm/cluster.py", line 3502, in node_setup
cl_inst.node_setup(_node, **setup_kwargs)
File "/jenkins/slave/workspace/scylla-staging/maciej/artifacts-ubuntu/scylla-cluster-tests/sdcm/cluster.py", line 4190, in node_setup
self._scylla_post_install(node, install_scylla, nic_devname)
File "/jenkins/slave/workspace/scylla-staging/maciej/artifacts-ubuntu/scylla-cluster-tests/sdcm/cluster.py", line 4266, in _scylla_post_install
node.scylla_setup(disks, devname)
File "/jenkins/slave/workspace/scylla-staging/maciej/artifacts-ubuntu/scylla-cluster-tests/sdcm/utils/decorators.py", line 113, in inner
res = func(*args, **kwargs)
File "/jenkins/slave/workspace/scylla-staging/maciej/artifacts-ubuntu/scylla-cluster-tests/sdcm/cluster.py", line 2225, in scylla_setup
self.remoter.run('sudo /usr/lib/scylla/scylla_setup --nic {} --disks {} --setup-nic-and-disks {}'
File "/jenkins/slave/workspace/scylla-staging/maciej/artifacts-ubuntu/scylla-cluster-tests/sdcm/remote/remote_base.py", line 613, in run
result = _run()
File "/jenkins/slave/workspace/scylla-staging/maciej/artifacts-ubuntu/scylla-cluster-tests/sdcm/utils/decorators.py", line 64, in inner
return func(*args, **kwargs)
File "/jenkins/slave/workspace/scylla-staging/maciej/artifacts-ubuntu/scylla-cluster-tests/sdcm/remote/remote_base.py", line 604, in _run
return self._run_execute(cmd, timeout, ignore_status, verbose, new_session, watchers)
File "/jenkins/slave/workspace/scylla-staging/maciej/artifacts-ubuntu/scylla-cluster-tests/sdcm/remote/remote_base.py", line 537, in _run_execute
result = connection.run(**command_kwargs)
File "/jenkins/slave/workspace/scylla-staging/maciej/artifacts-ubuntu/scylla-cluster-tests/sdcm/remote/libssh2_client/__init__.py", line 620, in run
return self._complete_run(channel, exception, timeout_reached, timeout, result, warn, stdout, stderr)
File "/jenkins/slave/workspace/scylla-staging/maciej/artifacts-ubuntu/scylla-cluster-tests/sdcm/remote/libssh2_client/__init__.py", line 654, in _complete_run
raise UnexpectedExit(result)
sdcm.remote.libssh2_client.exceptions.UnexpectedExit: Encountered a bad command exit code!
Command: 'sudo /usr/lib/scylla/scylla_setup --nic ens5 --disks /dev/nvme0n1 --setup-nic-and-disks  --swap-directory / '
Exit code: 1
Stdout:
#4  0x000055cb6c130afd execute_command_internal (bash + 0x4dafd)
#5  0x000055cb6c133158 execute_command (bash + 0x50158)
#6  0x000055cb6c12438b reader_loop (bash + 0x4138b)
#7  0x000055cb6c115c06 main (bash + 0x32c06)
#8  0x00007f891239dd90 n/a (libc.so.6 + 0x29d90)
#9  0x00007f891239de40 __libc_start_main (libc.so.6 + 0x29e40)
#10 0x000055cb6c115ed5 _start (bash + 0x32ed5)
Does not able to detect coredump file, failed to configure systemd-coredump.
coredump setup failed.
Stderr:
INFO  2022-05-26 11:14:32,107 [shard 0] seastar - Created fair group io-queue-0, capacity rate 2147483:2147483, limit 12582912, rate 16777216 (factor 1), threshold 2000
INFO  2022-05-26 11:14:32,107 [shard 0] seastar - Created io group dev(0), length limit 4194304:4194304, rate 2147483647:2147483647
INFO  2022-05-26 11:14:32,107 [shard 0] seastar - Created io queue dev(0) capacities: 512:2000/2000 1024:3000/3000 2048:5000/5000 4096:9000/9000 8192:17000/17000 16384:33000/33000 32768:65000/65000 65536:129000/129000 131072:257000/257000
WARN  2022-05-26 11:14:32,187 [shard 0] iotune - Available space on filesystem at /var/tmp/mnt: 121 MB: is less than recommended: 10 GB
INFO  2022-05-26 11:14:32,187 [shard 0] iotune - /var/tmp/mnt passed sanity checks
Created symlink /etc/systemd/system/multi-user.target.wants/scylla-server.service → /lib/systemd/system/scylla-server.service.
Created symlink /etc/systemd/system/multi-user.target.wants/var-lib-scylla.mount → /etc/systemd/system/var-lib-scylla.mount.
Created symlink /etc/systemd/system/multi-user.target.wants/var-lib-systemd-coredump.mount → /etc/systemd/system/var-lib-systemd-coredump.mount.
```
**<<<<<<<**

## Restore commands:
Restore Monitor Stack command: $ hydra investigate show-monitor 0d493f3f-2084-4689-b59b-783aeab5ff8c
Restore monitor on AWS instance using [Jenkins job](https://jenkins.scylladb.com/view/QA/job/QA-tools/job/hydra-show-monitor/parambuild/?sct_branch=add-ubuntu22-artifacts-test&test_id=0d493f3f-2084-4689-b59b-783aeab5ff8c)
Show all stored logs command: $ hydra investigate show-logs 0d493f3f-2084-4689-b59b-783aeab5ff8c

## Logs:
[db-cluster - https://cloudius-jenkins-test.s3.amazonaws.com/0d493f3f-2084-4689-b59b-783aeab5ff8c/20220526_111811/db-cluster-0d493f3f.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/0d493f3f-2084-4689-b59b-783aeab5ff8c/20220526_111811/db-cluster-0d493f3f.tar.gz)
[sct - https://cloudius-jenkins-test.s3.amazonaws.com/0d493f3f-2084-4689-b59b-783aeab5ff8c/20220526_111811/sct-runner-0d493f3f.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/0d493f3f-2084-4689-b59b-783aeab5ff8c/20220526_111811/sct-runner-0d493f3f.tar.gz)

## Links:
[Build URL](https://jenkins.scylladb.com/job/scylla-staging/job/maciej/job/artifacts-ubuntu/40/)

**Comment by syuu1228** [comment]
Seems like this is because coredumpctl changed output format (again).

Before v248, the output format was like this:
 - Storage: none
 - Storage: journal
 - Storage: /path/to/file (inacessible)
 - Storage: /path/to/file
However, after v248 it changed as follows:
 - Storage: none
 - Storage: journal
 - Storage: /path/to/file (inacessible)
 - Storage: /path/to/file (present)

So we have to truncate ` (present)` from the output.

see: https://github.com/systemd/systemd/blob/v248/src/coredump/coredumpctl.c#L703

**Comment by slivne** [comment]
@roydahan - do we have tests passing now on 22.04 ?

**Comment by roydahan** [comment]
No, there were no new runs since this fixed was merged.
@yaronkaikov, can you please check if this job is triggered correctly?

**Comment by yaronkaikov** [comment]
https://github.com/scylladb/scylla-pkg/pull/2935 was open, under review

**Comment by roydahan** [comment]
@yaronkaikov can you please trigger a build for ubuntu so we can test ubuntu22.04 with the fix?
The last one we tested is from 20220612 and it failed, so I don't know if we should reopen the issue or not.

**Comment by yaronkaikov** [comment]
@roydahan The name of the test was wrong so we couldn't find it (during the trigger) i fixed it and running it again, it should trigger in the next hour

**Comment by fgelcer** [comment]
> @roydahan The name of the test was wrong so we couldn't find it (during the trigger) i fixed it and running it again, it should trigger in the next hour

@KnifeyMoloko FYI

**Comment by yaronkaikov** [comment]
@roydahan https://jenkins.scylladb.com/job/scylla-master/job/artifacts/job/artifacts-ubuntu2204-arm-test/ passed but https://jenkins.scylladb.com/job/scylla-master/job/artifacts/job/artifacts-ubuntu2204-test/ didn't (looks like it never passed)

**Comment by fgelcer** [comment]
@yaronkaikov @syuu1228 , this fix didn't reach 5.0
i will now check master if it still happens

**Comment by yaronkaikov** [comment]
@avikivity Can you please backport this to 5.0 and then 2022.1 ?

**Comment by fgelcer** [comment]
@avikivity @slivne , without this fix (ad2344a864a348f77fc2b217ee18964cfdc32e1e) we cannot support Ubuntu 22.04 for 5.0 and 2022.1, FYI

**Comment by avikivity** [comment]
While not a regression, it's important to support new OS distributions. Backported to 4.6, 5.0.
